File: src/site/js/models/musica.js
class Musica {

    constructor(id, titulo, tom, tempo, secoes) {
        this.id = id
        this.titulo = titulo
        this.tom = tom
        this.tempo = tempo
        this.secoes = secoes
    }

    subirTom() {
        let notas = ['C', 'C#', 'D', 'D#', 'E', 'F', 'F#', 'G', 'G#', 'A', 'A#', 'B']
        if (this.tom == 'B') {
            this.tom = 'C'
            return
        }
        let index = notas.indexOf(this.tom)
        this.tom = notas[index + 1]

    }

}

var musicas2 = [
    {
        titulo: 'Santo Espirito',
        tom: 'E',
        tempo: 90,
        acordes: [
            'E|8', 'A|8',
        ]
    },
    new Musica('Santo Espirito', 'E', 90,
        [
            new Secao('Intro',
                [
                    Acorde.criarAcorde('E7(9)11+|8'),
                    Acorde.criarAcorde('A4|8'),
                ])
        ])

]

File: src/Cifralite.Web/Core/Entities/Musica.cs
namespace Cifralite.Web.Core.Entities;


public class Musica
{
    public readonly static string[] Tons = new[] { "C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B" };

    public int Id { get; set; }
    public string Titulo { get; set; } = string.Empty;
    public string Artista { get; set; } = string.Empty;
    public int Tempo { get; set; }
    public string? Tom { get; internal set; }
    public Usuario Usuario { get; set; } = new();

    public List<Secao> Secoes { get; set; } = new();

    public void DescerMeioTom()
    {
        if (Tom == "C")
        {
            Tom = "B";
        }
        else
        {
            Tom = Tons[Array.IndexOf(Tons, Tom) - 1];
        }

        foreach (var acorde in Secoes.SelectMany(x => x.Acordes))
        {
            acorde.DescerMeioTom();
        }
    }

    public void SubirMeioTom()
    {
        if (Tom == "B")
        {
            Tom = "C";
        }
        else
        {
            Tom = Tons[Array.IndexOf(Tons, Tom) + 1];
        }

        foreach (var acorde in Secoes.SelectMany(x => x.Acordes))
        {
            acorde.SubirMeioTom();
        }
    }
}

